internal/artifacts/module_arch.go

```diff
@@ -115,6 +120,48 @@ func ExecuteSoloBuild(source, target string, extensions []string, modulesNames [
 	return nil
 }
 
+func checkBuildResultsConflicts(mtaObj *mta.MTA, source, target string, extensions []string, modulesNames []string, wdGetter func() (string, error)) error {
+
+	modulesToPack := make(map[string]bool)
+	resultPathModuleNameMap := make(map[string]string)
+	for _, moduleName := range modulesNames {
+		_, err := mtaObj.GetModuleByName(moduleName)
+		if err != nil {
+			return err
+		}
+		modulesToPack[moduleName] = true
+	}
+
+	for _, module := range mtaObj.Modules {
+
+		moduleLoc, err := getModuleLocation(source, target, module.Name, extensions, wdGetter)
+		if err != nil {
+			return err
+		}
+
+		if modulesToPack[module.Name] {
+			_, defaultBuildResult, err := commands.CommandProvider(*module)
+			if err != nil {
+				return err
+			}
+			targetArtifact, _, err := buildops.GetModuleTargetArtifactPath(moduleLoc, false, module, defaultBuildResult, false)
+			if err != nil {
+				return err
+			}
+			if strings.ContainsAny(targetArtifact, "*?[]") {
+				continue
+			}
+			moduleName, pathInUse := resultPathModuleNameMap[targetArtifact]
+			if pathInUse {
+				return errors.Errorf(multiBuildWithPathsConflictMsg, module.Name, moduleName, moduleLoc.GetTarget(), filepath.Base(targetArtifact))
```

Comment:
final agreement was on this (see e-mails):
_could not build the selected modules: could not save the build results of modules "n2" and "n1" in the specified target folder (C:\Users......\testpartialbuild\partialbuildtest1) because of conflicting naming (data.zip); use the "build-artifact-name" build parameter to create a unique name for each module’s build result or use the default target folder._

Reply:
Yes, but we said we'll take the folder from `targetArtifact` like we take the file name from it

Reply:
done, I restored the code not from the latest version :-(